src/lib/validate-forms.js

```diff
@@ -0,0 +1,67 @@
+const api = require('./api')();
+const argsFormFilter = require('./args-form-filter');
+const fs = require('./sync-fs');
+const log = require('./log');
+const { getFormDir, getFormFilePaths } = require('./forms-utils');
+
+//TODO Add this to the main loop
+module.exports = (projectDir, subDirectory, options) => {
+  if (!options) options = {};
+
+  const formsDir = getFormDir(projectDir, subDirectory);
+  if(!formsDir) {
+    log.warn(`Forms dir not found: ${formsDir}`);
+    return Promise.resolve();
+  }
+
+  let hasInstanceIdFoundErrors = false;
+  let validateEndpointFoundErrors = false;
+  return argsFormFilter(formsDir, '.xml', options)
+    .reduce((promiseChain, fileName) => {
```

Comment:
Avoid `reduce` where possible. It's unnecessarily hard to read and provides no benefit over alternatives. In this case I think you can just use a loop with an await for the async bit and it'll be much more readable!

Reply:
Yes actually I did struggle with this reduce a bit, it is a copy and paste of how update-forms.js works, I can refactor both.